### PR TITLE
Fix the JSON layout logging levels mapping

### LINKED ISSUE #299: Logging: warning level is incorrectly reported to Cloud Logging
Following the [documentation](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#LogSeverity), the `severity` field of a log entry must be one of the possbile nine options described.

However, this is being reported as a [direct translation](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/0fe8d71268da4fd1def6c02868b25ecafe6855c4/spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java#L216) of the `Level` enumeration.

As such, the following happens:

| `Level` | `LogSeverity` | Effect in Cloud Logging |
|---------------|--------|------------------------|
| TRACE | ? | DEFAULT ❌ |
| DEBUG | DEBUG | DEBUG ✅ |
| INFO | INFO | INFO ✅ |
| WARN | WARNING | DEFAULT ❌ |
| ERROR | ERROR | ERROR ✅ |

I guess the most pressing of these is the warning level translation.

### PR DESCRIPTION
**Pull request by dzou**
Fix the Logback to Severity mapping in the `StackdriverJsonLayout`.

Fixes #299.

### CODE REVIEW

File: spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java [hunk 1]
```diff
@@ -212,8 +224,11 @@ public void start() {
 					TimeUnit.MILLISECONDS.toNanos(event.getTimeStamp() % 1_000));
 		}
 
-		add(StackdriverTraceConstants.SEVERITY_ATTRIBUTE, this.includeLevel,
-				String.valueOf(event.getLevel()), map);
+		add(StackdriverTraceConstants.SEVERITY_ATTRIBUTE,
+				this.includeLevel,
+				logbackToSeverityMap.getOrDefault(event.getLevel(), Severity.DEFAULT.name()),
```
Comment (ttomsu) [inline spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java:229]: Would you mind adding a small test to exercise the `orDefault` part?

**Review by ttomsu — APPROVED** [review]

**Review by meltsufin — APPROVED** [review]
Optional: add more coverage for all logging levels, especially TRACE.

File: spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java (hunk 1 above)
Comment (dzou) [reply spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/StackdriverJsonLayout.java:228]: Done.

**Comment by dzou** [comment]
I added the test `testJsonSeverityLevelMapping()` and ended up refactoring the test significantly. Should be easier to add tests in the future. Unfortunately there doesn't seem to be a way to trigger the `DEFAULT` via the logger for the unit test..

**Review by meltsufin — APPROVED** [review]
Nice work!

**Pull request merged**